mcs: update gc safe point doesn't work normally

## Bug Report

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What did you do?

<!-- If possible, provide a recipe for reproducing the error. -->
`~/pingcap/tiup/bin/tiup-playground v6.6.0  --pd.micro '[{"api":true, "amount":1},{"tso":true,"amount":1}]' --pd.binpath ~/pd-cse/bin/pd-server --db.binpath ~/tidb-cse/bin/tidb-server --kv 3 --db 1 --tiflash 0 --db.port 14000 --host 172.16.4.3`
### What did you expect to see?

### What did you see instead?
`[2023/03/31 18:18:00.646 +08:00] [ERROR] [gc_worker.go:744] ["[gc worker] failed to update service safe point"] [uuid=61cdc8da8100014] [error="rpc error: code = DeadlineExceeded desc = context deadline exceeded"]`
### What version of PD are you using (`pd-server -V`)?
master

--- a/server/grpc_service.go
+++ b/server/grpc_service.go
@@ -1766,7 +1766,7 @@ func (s *GrpcServer) getGlobalTSOFromTSOServer(ctx context.Context) (pdpb.Timest
 	if !ok || forwardedHost == "" {
 		return pdpb.Timestamp{}, ErrNotFoundTSOAddr
 	}
-	forwardStream, err := s.getTSOForwardStream(ctx, forwardedHost)
+	forwardStream, err := s.getTSOForwardStream(forwardedHost)
 	if err != nil {
 		return pdpb.Timestamp{}, err
 	}
@@ -1780,12 +1780,13 @@ func (s *GrpcServer) getGlobalTSOFromTSOServer(ctx context.Context) (pdpb.Timest
 	})
 	ts, err := forwardStream.Recv()
 	if err != nil {
+		log.Error("get global tso from tso server failed", zap.Error(err))
 		return pdpb.Timestamp{}, err
 	}
 	return *ts.GetTimestamp(), nil
 }
 
-func (s *GrpcServer) getTSOForwardStream(ctx context.Context, forwardedHost string) (tsopb.TSO_TsoClient, error) {
+func (s *GrpcServer) getTSOForwardStream(forwardedHost string) (tsopb.TSO_TsoClient, error) {
 	s.tsoClientPool.RLock()
 	forwardStream, ok := s.tsoClientPool.clients[forwardedHost]
 	s.tsoClientPool.RUnlock()
@@ -1804,18 +1805,18 @@ func (s *GrpcServer) getTSOForwardStream(ctx context.Context, forwardedHost stri
 	}
 
 	// Now let's create the client connection and the forward stream
-	client, err := s.getDelegateClient(ctx, forwardedHost)
+	client, err := s.getDelegateClient(s.ctx, forwardedHost)
 	if err != nil {
 		return nil, err
 	}
 	done := make(chan struct{})
-	ctx, cancel := context.WithTimeout(s.ctx, tsoutil.DefaultTSOProxyTimeout)
+	ctx, cancel := context.WithCancel(s.ctx)
 	go checkStream(ctx, cancel, done)
 	forwardStream, err = tsopb.NewTSOClient(client).Tso(ctx)
+	done <- struct{}{}
 	if err != nil {
 		return nil, err
 	}
-	done <- struct{}{}
 	s.tsoClientPool.clients[forwardedHost] = forwardStream
 	return forwardStream, nil
 }

